cfc-testing/scripts/deployment-dry-run-tracker.sh

```
@@ -0,0 +1,437 @@
+
+        echo -e "  ${RED}❌ Synthesis failed (${synth_duration}s)${NC}" | tee -a "$REPORT_FILE"
+        local error_msg=$(head -1 "$synth_error" | tr ',' ' ')
+        echo "  Error: $error_msg" | tee -a "$REPORT_FILE"
+
+        # Record failure
+        echo "$RUN_ID,$(date '+%Y-%m-%d %H:%M:%S'),$stack_name,$runtime,$security_profile,$auth_mode,$network_mode,$compliance_frameworks,$synth_duration,0,0,SYNTH_FAILED,$error_msg" >> "$METRICS_CSV"
+
+        return 1
+    fi
+
+    local synth_end=$(date +%s.%N)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)